dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/webcontainerext/JSPExtensionFactory.java

```diff
@@ -598,7 +608,12 @@ public static synchronized GlobalTagLibraryCache getGlobalTagLibraryCache() {
      */
     @Reference(cardinality=ReferenceCardinality.MULTIPLE, policy=ReferencePolicy.DYNAMIC)
     protected void setGlobalTagLibConfig(GlobalTagLibConfig globalTagLibConfig) {
-            getGlobalTagLibraryCache().addGlobalTagLibConfig(globalTagLibConfig);
+            // Wait for pages version to be loaded first before calling getGlobalTagLibraryCache
+            if(!isPagesVersionLoaded){
+                storedGlobalTagLibConfigs.add(globalTagLibConfig);
+            } else {
+                getGlobalTagLibraryCache().addGlobalTagLibConfig(globalTagLibConfig);
+            }
```

Comment:
_setVersion_ is called after _setGlobalTagLibConfig_, so we just store the globalTagLibConfig and then add it to the GlobalTagLibraryCache in the setVersion